azure/converters/vm.go

```diff
@@ -61,7 +61,7 @@ func SDKToVM(v compute.VirtualMachine) *VM {
 		vm.AvailabilityZone = to.StringSlice(v.Zones)[0]
 	}
 
-	if len(v.Tags) > 0 {
+	if v.Tags != nil && len(v.Tags) > 0 {
```

Comment:
why do we need this extra check? a nil v.Tags will already not pass the length check https://go.dev/play/p/QoprOQrAlzV